gemini/agent-engine/tutorial_ag2_on_agent_engine.ipynb

```diff
@@ -0,0 +1,939 @@
+        "\n",
+        "In many cases, running `gcloud auth application-default login` in a shell on the machine running the notebook kernel is sufficient.\n",
+        "\n",
+        "More authentication options are discussed [here](https://cloud.google.com/docs/authentication)."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 2,
+      "metadata": {
+        "cellView": "form",
+        "id": "rPZA1RYSj0Vs"
+      },
+      "outputs": [],
+      "source": [
+        "# @title Authenticate - Google Colab Only\n",
+        "import sys\n",
+        "\n",
+        "from google.colab import auth\n",
+        "\n",
+        "if \"google.colab.auth\" in sys.modules:\n",
+        "    auth.authenticate_user()\n",
+        "    print(\"Authenticated\")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "jQs3DQ4DxB_k"
+      },
+      "source": [
+        "###  Install Dependencies and Import Packages"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 3,
+      "metadata": {
+        "cellView": "form",
+        "id": "1LCcbgeYqXx6"
+      },
+      "outputs": [],
+      "source": [
+        "# @title Install/Upgrade Vertex AI SDK - Restart Runtime (If needed)\n",
+        "\n",
+        "\n",
+        "def install_or_upgrade_vertex_ai():\n",
+        "    package_name = \"google-cloud-aiplatform\"\n",
+        "    minimum_version = \"1.83.0\"\n",
+        "\n",
+        "    try:\n",
+        "        import google.cloud.aiplatform as aiplatform\n",
+        "\n",
+        "        installed_version = aiplatform.__version__\n",
+        "\n",
+        "        if installed_version < minimum_version:\n",
+        "            print(\n",
+        "                f\"Upgrading {package_name} from version {installed_version} to {minimum_version}...\"\n",
+        "            )\n",
+        "            !pip install google-cloud-aiplatform --upgrade --quiet --user\n",
+        "            print(f\"Successfully upgraded {package_name}.\")\n",
+        "            restart_runtime()\n",
+        "        else:\n",
+        "            print(\n",
+        "                f\"{package_name} is already installed with version {installed_version}.\"\n",
+        "            )\n",
+        "    except ImportError:\n",
+        "        print(\n",
+        "            f\"{package_name} is not installed. Installing version {minimum_version}...\"\n",
+        "        )\n",
+        "        !pip install google-cloud-aiplatform --upgrade --quiet --user\n",
```

Comment:
Use `%pip install` to not need the Kernel Restart.

Follow the template here: https://github.com/GoogleCloudPlatform/generative-ai/blob/main/notebook_template.ipynb